guava/src/com/google/common/util/concurrent/Uninterruptibles.java

```diff
@@ -326,7 +326,67 @@ public static boolean tryAcquireUninterruptibly(
     }
   }
 
-  // TODO(user): Add support for waitUninterruptibly.
+  /**
+   * Invokes {@code object.}{@link Object#wait() wait()} uninterruptibly.
+   *
+   * @param object to wait upon.
+   *
+   * @throws IllegalMonitorStateException if the current thread is not
+   *               the owner of the object's monitor.
+   */
+  @GwtIncompatible // concurrency
+  public static void waitUninterruptibly(Object object) {
+    Preconditions.checkNotNull(object);
+    boolean interrupted = false;
+    try {
+      while (true) {
+        try {
+          object.wait();
+          return;
+        } catch (InterruptedException e) {
+          interrupted = true;
```

Comment:
That seems like a reasonable API. I think the main question is whether we want to get in the business of supplying high-level wrappers around `Object.wait` or whether we want our high-level APIs to be based on other primitives, like [`Monitor`](http://google.github.io/guava/releases/19.0/api/docs/com/google/common/util/concurrent/Monitor.html) or even `ReentrantLock`.

I'll see if I can get some other opinions on this.